src/FindApprenticeshipTraining/SFA.DAS.FindApprenticeshipTraining/InnerApi/Requests/GetCourseProviderDetailsRequest.cs

```diff
@@ -0,0 +1,57 @@
+﻿using SFA.DAS.SharedOuterApi.Interfaces;
+using System;
+using System.Web;
+
+namespace SFA.DAS.FindApprenticeshipTraining.InnerApi.Requests;
+
+public sealed class GetCourseProviderDetailsRequest : IGetApiRequest
+{
+    private int LarsCode { get; }
+    private long Ukprn { get; }
+    private string Location { get; }
+    private decimal? Longitude { get; }
+    private decimal? Latitude { get; }
+    private Guid? ShortlistUserId { get; }
+
+    public string GetUrl => BuildUrl();
+
+    public GetCourseProviderDetailsRequest(int larsCode, long ukprn, string location, decimal? longitude, decimal? latitude, Guid? shortlistUserId)
+    {
+        LarsCode = larsCode;
+        Ukprn = ukprn;
+        Location = location;
+        Longitude = longitude;
+        Latitude = latitude;
+        ShortlistUserId = shortlistUserId;
+    }
+
+    private string BuildUrl()
+    {
+        var query = HttpUtility.ParseQueryString(string.Empty);
+
+        if (!string.IsNullOrWhiteSpace(Location))
+        {
+            query["location"] = Location;
```

Comment:
this is a curious way to do this, did you find this pattern in use elsewhere? Does it auto deal with escape the location string?  If so leave as is, but I'm assuming you checked that odd characters (the sort hackers might try) are dealt with for location?